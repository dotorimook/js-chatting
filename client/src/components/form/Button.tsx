import React, { FC } from 'react'
import IComponentProps from 'interfaces/IComponentProps';
import styled from 'styled-components';
import theme from 'style/theme';
import Icon from 'components/Icon';

interface IButtonProps extends IComponentProps{
  icon?:string;
}

const ButtonStyled = styled.button`
  width: 100%;
  font-size:1em;
  border:none;
  padding: 1em 1em;
  background-color:transparent;

  .material-icons {
    vertical-align:bottom;
    &.left {
      margin-right:.2em;
    }
  }
  
  transition: all 0.2s ease;
  &:hover {
    background-color:${theme.color3}
  }
  &:active {
    transform:translateY(2px);
  }
  
  &[type=submit] {
    background-color:${theme.color3};
    &:hover {
      background-color:${theme.color2};
    }
  }
`

const Button:FC<IButtonProps>= (props:IButtonProps) => {
  return (
    <ButtonStyled {...(props as any)}>
      {
        props.icon && <Icon className={`${props.children?'left': ''}`} name={props.icon}/>
      }
      {props.children}
    </ButtonStyled>
  )
}

export default Button;