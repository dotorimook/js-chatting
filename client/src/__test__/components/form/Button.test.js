import React from 'react'
import 'setupTest';
import Button from 'components/form/Button';
import {shallow} from 'enzyme';

describe('Button Test', () => {
  let ui = null;
  const buttonText = 'button';

  beforeAll(()=>{
    ui = shallow(<Button onClick={()=>{alert('test');}}>{buttonText}</Button>);
  });

  it('matches snapshot', () => {
    expect(ui).toMatchSnapshot();
  });

  it('property render child', () => {
    expect(ui.text()).toEqual(buttonText);
  });

  it('properly onClick called', ()=> {
    window.alert = jest.fn();
    ui.simulate('click');
    expect(window.alert).toHaveBeenCalledWith('test');
  });
});