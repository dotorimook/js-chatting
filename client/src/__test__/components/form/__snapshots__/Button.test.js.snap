// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test matches snapshot 1`] = `
<styled.button
  onClick={[Function]}
>
  button
</styled.button>
`;
